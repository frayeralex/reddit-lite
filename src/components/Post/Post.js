import React from 'react';
import PropTypes from 'prop-types';
import m from 'moment';
import './Post.scss';

const Post = ({ data, moment }) => {
  return (
    <li className="Post">
      <header className="header-container">
        <div className="author">
          Posted by{' '}
          <a
            href={`https://www.reddit.com/user/${data.author}`}
            className="external-link"
            target="_blank"
            rel="noopener noreferrer"
          >
            u/{data.author} <i className="fa fa-external-link" />
          </a>{' '}
        </div>
        <span className="created-time">
          {moment(Number(`${data.created_utc}000`), 'x').fromNow()}
        </span>
      </header>
      <div className="middle-container">
        {data.thumbnail && data.thumbnail.includes('http') && (
          <div className="thumbnail-container">
            <img src={data.thumbnail} alt="" />
          </div>
        )}
        <div className="title">
          {data.title}{' '}
          <a
            href={data.url}
            className="external-link"
            target="_blank"
            rel="noopener noreferrer"
          >
            <i className="fa fa-external-link" />
          </a>
        </div>
      </div>
      <footer className="footer-container">
        <span className="comments-link">
          <a
            href={`https://www.reddit.com${data.permalink}`}
            className="external-link"
            target="_blank"
            rel="noopener noreferrer"
          >
            {data.num_comments} <i className="fa fa-comments" />
          </a>
        </span>
      </footer>
    </li>
  );
};

Post.defaultProps = {
  moment: m,
};

Post.propTypes = {
  data: PropTypes.object.isRequired,
  moment: PropTypes.func,
};

export default React.memo(Post);
