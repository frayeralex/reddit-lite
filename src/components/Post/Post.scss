.Post {
  margin: 0 auto;
  max-width: 900px;
  padding: 30px;
  background-color: rgba(0, 0, 0, 0.1);
  border: 1px solid rgba(0, 0, 0, 0.1);
  background-color: #fff;
  text-align: left;
  list-style: none;

  a {
    color: inherit;
    text-decoration: none;
  }

  .header-container {
    margin-bottom: 10px;
    display: flex;
  }

  .middle-container {
    display: flex;
  }

  .footer-container {
    display: flex;
  }

  .author,
  .created-time {
    color: rgb(120, 124, 126);
    font-size: 12px;
    line-height: 16px;
    font-weight: 600;
  }

  .author {
    flex-grow: 1;
  }

  .external-link {
    color: inherit;
  }

  .title {
    font-size: 18px;
    color: rgb(120, 124, 126);
    margin-bottom: 20px;

    .external-link {
      font-size: 20px;
    }
  }

  .self-text {
    font-size: 12px;
    color: rgb(120, 124, 126);
    margin-bottom: 10px;
  }

  .comments-link {
    color: rgb(120, 124, 126);
    font-size: 12px;
  }
}
