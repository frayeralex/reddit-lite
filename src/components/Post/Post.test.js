import React from 'react';
import Post from './Post';
import posts from '__fixtures__/posts';

jest.mock('moment', () => () => ({ fromNow: () => '16 hours ago' }));

describe('<Post/>', () => {
  it('should render without crushing ', () => {
    const [{ data }] = posts;
    const component = shallow(<Post data={data} />);
    expect(component).toMatchSnapshot();
  });
  it('should render thumbnail if it exist', function() {
    const { data } = posts.find(
      item => item.data.thumbnail && item.data.thumbnail.includes('http'),
    );
    const component = shallow(<Post data={data} />);
    expect(component).toMatchSnapshot();
  });
  it('should passed correct data to moment function and call fromNow', function() {
    const fromNow = jest.fn(() => '16 hours ago');
    const moment = jest.fn(() => ({ fromNow }));
    const [{ data }] = posts;
    shallow(<Post data={data} moment={moment} />);
    expect(moment).toHaveBeenCalledWith(Number(`${data.created_utc}000`), 'x');
    expect(fromNow).toHaveBeenCalled();
  });
});
