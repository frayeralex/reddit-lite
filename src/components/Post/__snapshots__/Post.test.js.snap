// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post/> should render thumbnail if it exist 1`] = `
<li
  className="Post"
>
  <header
    className="header-container"
  >
    <div
      className="author"
    >
      Posted by
       
      <a
        className="external-link"
        href="https://www.reddit.com/user/renaissancetroll"
        rel="noopener noreferrer"
        target="_blank"
      >
        u/
        renaissancetroll
         
        <i
          className="fa fa-external-link"
        />
      </a>
       
    </div>
    <span
      className="created-time"
    >
      16 hours ago
    </span>
  </header>
  <div
    className="middle-container"
  >
    <div
      className="thumbnail-container"
    >
      <img
        alt=""
        src="https://a.thumbs.redditmedia.com/stFTqRDLe3EBfpL5FF34QPvmFrMjKtmuKTxFlFrh374.jpg"
      />
    </div>
    <div
      className="title"
    >
      Twitch API Dashboard - A Fun, Extendable Beginner ReactJS Tutorial
       
      <a
        className="external-link"
        href="https://youtu.be/VTY6ZzDTV3A"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fa fa-external-link"
        />
      </a>
    </div>
  </div>
  <footer
    className="footer-container"
  >
    <span
      className="comments-link"
    >
      <a
        className="external-link"
        href="https://www.reddit.com/r/reactjs/comments/cmcq55/twitch_api_dashboard_a_fun_extendable_beginner/"
        rel="noopener noreferrer"
        target="_blank"
      >
        15
         
        <i
          className="fa fa-comments"
        />
      </a>
    </span>
  </footer>
</li>
`;

exports[`<Post/> should render without crushing  1`] = `
<li
  className="Post"
>
  <header
    className="header-container"
  >
    <div
      className="author"
    >
      Posted by
       
      <a
        className="external-link"
        href="https://www.reddit.com/user/swyx"
        rel="noopener noreferrer"
        target="_blank"
      >
        u/
        swyx
         
        <i
          className="fa fa-external-link"
        />
      </a>
       
    </div>
    <span
      className="created-time"
    >
      16 hours ago
    </span>
  </header>
  <div
    className="middle-container"
  >
    <div
      className="title"
    >
      Whoâ€™s Hiring? [August 2019]
       
      <a
        className="external-link"
        href="https://www.reddit.com/r/reactjs/comments/ckow24/whos_hiring_august_2019/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fa fa-external-link"
        />
      </a>
    </div>
  </div>
  <footer
    className="footer-container"
  >
    <span
      className="comments-link"
    >
      <a
        className="external-link"
        href="https://www.reddit.com/r/reactjs/comments/ckow24/whos_hiring_august_2019/"
        rel="noopener noreferrer"
        target="_blank"
      >
        46
         
        <i
          className="fa fa-comments"
        />
      </a>
    </span>
  </footer>
</li>
`;
